var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -156,6 +156,20 @@ def configure_args(self):
 
         return ["--with-zlib=%s" % spec['zlib'].prefix] + extra_args
 
+    def configure(self, spec, prefix):
+        # Run the default autotools package configure
+        super(Hdf5, self).configure(spec, prefix)
+
+        if '@:1.8.14' in spec:
```

Comment:
Oh cool.  I remember seeing this but it didn't come to mind because of the names, like you say.  I guess I was thinking it would be simpler if all the user had to write was something like:

```python
def post_configure(self, spec, prefix):
   # do something
```

Where there would already be a no-op hook implemented in the package superclass.  The build pipelin could look for functions called `pre_` and `post_<name>`.  The decorator is nice for sanity checks because you can separate them out by functions, but it is not all that readable...

Also what if the sanity check decorator were a free function?  e.g.:

```python
@run_after('configure')
def ubuntu_fix(self):
    pass
```

Seems like we shouldn't have to specify `AutotoolsPackage` here.